Detectors/GlobalTracking/src/MatchTOF.cxx

```diff
@@ -386,9 +497,143 @@ bool MatchTOF::prepareTracks()
+      auto& trcB = mTracksWork[b].second;
+      return ((trcA.getTimeStamp() - mSigmaTimeCut * trcA.getTimeStampError()) - (trcB.getTimeStamp() - mSigmaTimeCut * trcB.getTimeStampError()) < 0.);
+    });
+  } // loop over tracks of single sector
+
+  // Uncomment for local debug
+  /* 
+  // printing the tracks
+  std::array<float, 3> globalPos;
+  int itmp = 0;
+  for (int sec = o2::constants::math::NSectors; sec--;) {
+    Printf("sector %d", sec);
+    auto& cacheTrk = mTracksSectIndexCache[sec];   // array of cached tracks indices for this sector; reminder: they are ordered in time!
+    for (int itrk = 0; itrk < cacheTrk.size(); itrk++){
+      itmp++; 
+      auto& trc = mTracksWork[cacheTrk[itrk]];
+      trc.getXYZGlo(globalPos);
+      printf("Track %d: Global coordinates After propagating to 371 cm: globalPos[0] = %f, globalPos[1] = %f, globalPos[2] = %f\n", itrk, globalPos[0], globalPos[1], globalPos[2]);
+      //      Printf("The phi angle is %f", TMath::ATan2(globalPos[1], globalPos[0]));
+    }
+  }
+  Printf("we have %d tracks",itmp);      
+  */
+
+  return true;
+}
+//______________________________________________
+bool MatchTOF::prepareTPCTracks()
+{
+  ///< prepare the tracks that we want to match to TOF
+
+  if (!mIsworkflowON && !loadTPCTracksNextChunk()) {
+    return false;
+  }
+
+  mNumOfTracks = mTPCTracksArrayInp.size();
+  if (mNumOfTracks == 0)
+    return false; // no tracks to be matched
+  mMatchedTracksIndex.resize(mNumOfTracks);
+  std::fill(mMatchedTracksIndex.begin(), mMatchedTracksIndex.end(), -1); // initializing all to -1
+
+  // copy the track params, propagate to reference X and build sector tables
+  mTracksWork.clear();
+  mTracksWork.reserve(mNumOfTracks);
+
+  for (int sec = o2::constants::math::NSectors; sec--;) {
+    mTPCTracksSectIndexCache[sec].clear();
+    mTPCTracksSectIndexCache[sec].reserve(100 + 1.2 * mNumOfTracks / o2::constants::math::NSectors);
+  }
+
+  // getting Bz (mag field)
+  auto o2field = static_cast<o2::field::MagneticField*>(TGeoGlobalMagField::Instance()->GetField());
+  float bzField = o2field->solenoidField(); // magnetic field in kGauss
+  float minpt = abs(bzField) * 0.05;
+  int nclustersMin = 0;
+  LOG(INFO) << "Min track pT allowed = " << minpt;
+  LOG(INFO) << "Min track Nclusters allowed = " << nclustersMin;
+
+  LOG(DEBUG) << "\n\nWe have %d tracks to try to match to TOF: " << mNumOfTracks;
+  int nNotPropagatedToTOF = 0;
+  for (int it = 0; it < mNumOfTracks; it++) {
+    const o2::tpc::TrackTPC& trcOrig = mTPCTracksArrayInp[it]; // TODO: check if we cannot directly use the o2::track::TrackParCov class instead of o2::dataformats::TrackTPCITS, and then avoid the casting below; this is the track at the vertex
+    std::array<float, 3> globalPos;
+
+    // create working copy of track param
+    timeEst timeInfo;
+    // set
+    timeInfo.setTimeStamp(trcOrig.getTime0());
+    timeInfo.setTimeStampError(trcOrig.getDeltaTBwd());
```

Comment:
Note the DeltaTBwd is not an error: ``{getTime0 - getDeltaTBwd() : getTime0 + getDeltaTBwd()} define allowed track times. Perhaps you can use instead of TimeStampWithError this class: https://github.com/AliceO2Group/AliceO2/blob/dev/Common/MathUtils/include/MathUtils/Bracket.h as ``Bracket<float> trange`` (provides min/max + comparison operators) ? But then this should be done also for global tracks

Reply:
I simply forced (now) mSigmaTimeCut = 1. Is it enough?

Reply:
no, with mSigmaTimeCut=1 you will get correct minTime, but not the maxTime: the TBwd and Fwd have nothing to do with errors, see their dependence in TPC timebins (200 ns)
![tFwdBwd](https://user-images.githubusercontent.com/7382029/94735008-5c9ad600-036a-11eb-9932-f557014678ed.gif)

Reply:
ok. done